packages/mui-base/src/utils/useScrollLock.ts

```diff
@@ -1,75 +1,110 @@
-import { useEnhancedEffect } from './useEnhancedEffect';
-import { useId } from './useId';
 import { isIOS } from './detectBrowser';
+import { useEnhancedEffect } from './useEnhancedEffect';
+
+let originalRootStyles = {};
+let originalBodyStyles = {};
+let preventScrollCount = 0;
+let restore: () => void = () => {};
+
+function preventScrollIOS() {
+  // To implement
+  return () => {};
+}
+
+function preventScrollStandard() {
+  const html = document.documentElement;
+  const body = document.body;
+  const rootStyle = html.style;
+  const bodyStyle = body.style;
+
+  let resizeRaf: number;
+  let scrollX: number;
+  let scrollY: number;
+
+  function lockScroll() {
+    const offsetLeft = window.visualViewport?.offsetLeft || 0;
+    const offsetTop = window.visualViewport?.offsetTop || 0;
+    const hasConstantOverflowY = rootStyle.overflowY === 'scroll';
+    const hasConstantOverflowX = rootStyle.overflowX === 'scroll';
+
+    scrollX = rootStyle.left ? parseFloat(rootStyle.left) : window.scrollX;
+    scrollY = rootStyle.top ? parseFloat(rootStyle.top) : window.scrollY;
+
+    originalRootStyles = {
+      position: rootStyle.position,
+      top: rootStyle.top,
+      left: rootStyle.left,
+      right: rootStyle.right,
+      overflowX: rootStyle.overflowX,
+      overflowY: rootStyle.overflowY,
+    };
+    originalBodyStyles = {
+      overflow: bodyStyle.overflow,
+    };
+
+    Object.assign(rootStyle, {
+      // Handle `scrollbar-gutter` in Chrome when there is no scrollable content.
+      position: html.scrollHeight > html.clientHeight ? 'fixed' : '',
+      top: `${-(scrollY - Math.floor(offsetTop))}px`,
+      left: `${-(scrollX - Math.floor(offsetLeft))}px`,
+      right: '0',
```

Comment:
- We don't need to set `bottom`?
- Nit: if we are setting position conditionally, should we also set top/left/right conditionally? If someone sets a `position` value on their `html` element, for whatever weird reason

Reply:
- `bottom` isn't necessary
- Changed